Release 1.0

mcupak opened the pull request:
Last chance for 1.0 requests before we tag from `master`. Please speak now or forever hold your peace.

mcupak committed:
Add service-info endpoint, fix #27

mcupak committed:
Correct spec info, fix #5

mcupak committed:
Add bearer auth support, fix #29

mcupak committed:
Add GA4GH logo to README, issue #25

mcupak committed:
Add view in  Swagger Editor link, issue #25

mcupak committed:
Add info on how to report security vulnerabilities, issue #25

mcupak committed:
Add standard CORS language, issue #25

mcupak committed:
Merge branch 'develop' into issue/25

mcupak committed:
Change framework to API in spec description, fix #5

mcupak committed:
Merge branch 'issue/5' into develop

mcupak committed:
Remove restriction on token format, fix #29

mcupak committed:
Rename tags to be consistent with service-info, fix #27

Commit:
Added error object and error attributes to response objects

Commit:
fix spelling mistakes

sharvari-kapadia committed:
Allow implementer of the GET /tables to extend the specification by returning additional properties.

This change:
* Allows to solve specific use cases that are difficult with the spec as written
* Allows to experiment with draft extensions to the upcoming versions of new specifications

sharvari-kapadia committed:
Remove previous_page_url from Pagination object and update the README.md to remove its reference.

Within the Pagination object, previous_page_url parameter is not useful. Hence, it is better to leave it out of the spec.

mcupak committed:
Merge pull request #40 from ga4gh-discovery/remove-previous_page_url

Remove previous_page_url from Pagination object

mcupak committed:
Merge pull request #39 from ga4gh-discovery/allow-additional-properties

Allow additional properties in table list specification

mcupak committed:
Merge develop

Commit:
converted discovery search specification document to md document

Commit:
added reference to full API specs in README.md

Commit:
fixes brokenTOC reference links

Commit:
Removes a small blurb that did not belong in this document

Commit:
Removed references to AUTH_CHALLENGE

Commit:
Added better documentation for source and error code sections

Commit:
Fixed spelling typo

Commit:
Phenopackets examples updated

sharvari-kapadia committed:
[#174845155]: Fix typo in the phenopacket table name.

Commit:
Fixed formatting and rebase errors

Commit:
updated with lates long running queries section

Commit:
broken format fix

Commit:
Merge pull request #42 from DNAstack/api-doc

Move search specification document from Google Docs to Markdown

Commit:
ports documentation to redoc

Commit:
Merge branch 'develop' of github.com:ga4gh-discovery/ga4gh-discovery-search into redoc

Commit:
merged ignore file

Commit:
move examples and use cases to the front in the readme file

Commit:
Updated links to point to openapi/openapi.yaml. Moved examples further up

mcupak committed:
Bump version to 0.2.0

mcupak committed:
Merge pull request #31 from mcupak/issue/27

Add service-info endpoint, fix #27

shiroyuki committed:
Update the error structure to be similar to Service Registry' error object

Commit:
removes specific reference to projects/deployments from examples

Commit:
Merge branch 'develop' of github.com:ga4gh-discovery/ga4gh-discovery-search into redoc
spec/search-api.yaml -> openapi/openapi.yaml

Commit:
added travis integration, remove the need to use npm

Commit:
rewords certain API descriptions to be more clear

Commit:
adds in missing footnote about route naming

shiroyuki committed:
Updated the spec based on the feedback with Miro and Jonathan[#175022439]

shiroyuki committed:
changed all 'default' to '500' [#175022439]

shiroyuki committed:
resolved merge conflict [#175022439]

shiroyuki committed:
Removed '500' from /service-info [#175022439]

shiroyuki committed:
Refactored the schema [#175022439]

jfuerth committed:
Initial draft for addressing issue #47

jfuerth committed:
Improve wording and clarity on parameterized queries (#47)

mcupak committed:
Add Apache voting system to contributing guidelines, issue #25

mcupak committed:
Rename Discovery Search to GA4GH Search, issue #25

mcupak committed:
Split example in SearchRequest under properties so that it can render in common editors, issue #47

mcupak committed:
Merge pull request #37 from DNAstack/add-errors

Add Error Object to Responses

mcupak committed:
Merge pull request #50 from mcupak/rename-search

Rename Discovery Search to GA4GH Search, issue #25

mcupak committed:
Merge pull request #48 from ga4gh-discovery/issue-47-search-request-body

Document query request  body and parameter types

mcupak committed:
Consolidate docs, issue #25

mcupak committed:
Merge branch 'develop' into consolidate-docs

mcupak committed:
Consider pull request review actions on GitHub votes in the Apache process, issue #25

mcupak committed:
Bump spec version to 1.0.0, issue #25

mcupak committed:
Merge pull request #49 from mcupak/add-apache-voting

Add Apache voting system to contributing guidelines, issue #25

mcupak committed:
Merge pull request #51 from mcupak/consolidate-docs

Consolidate docs

mcupak committed:
Merge branch 'develop' into release-1.0.0

Commit:
Merge branch 'develop' of github.com:ga4gh-discovery/ga4gh-search into redoc

mcupak committed:
Remove broken link to tables-in-a-bucket docs, fix #52

sharvari-kapadia committed:
[#174957722], [#175522776]: Credit original sources of phenopacket examples & update the phenopacket Storage API links.

mcupak committed:
Merge pull request #54 from ga4gh-discovery/update-phenopacket-section

Credit original sources of phenopacket examples & update the phenopacket Storage API links.

mcupak committed:
Merge pull request #53 from mcupak/issue/52

Remove broken link to tables-in-a-bucket docs, fix #52

Commit:
Merge branch 'develop' into redoc

Commit:
Merge pull request #46 from DNAstack/redoc

ports documentation to redoc

ianfore committed:
pre prc

ianfore committed:
Update README.md

ianfore committed:
Suggestion for PRC release - IF Final

ianfore committed:
Added second example use case to summary

mcupak committed:
Fix links in docs

mcupak committed:
Spellchecked SEARCHSPEC.md

mcupak committed:
Add REWS security clause to README, fix #64

mcupak committed:
Fix meeting minutes and contact links in README, issue #64

mcupak committed:
Plug in implementation docs, fix #69

mcupak committed:
Merge pull request #65 from mcupak/issue/64

Add REWS security clause to README, fix #64

Commit:
Merges in Hugo docs (#2)

Adds Hugo to Search Spec, without touching source.

Commit:
removes built source and adds clean script for makefile

Commit:
Keep history should be kept off for github pages deployment, because it will cause merge conflicts

Commit:
removes broken git modules command

ianfore committed:
typos and DRS addition

mcupak committed:
Merge pull request #77 from ianfore/release-1.0.0

typos and DRS addition

mcupak committed:
Merge pull request #72 from mcupak/issue/69

Plug in implementation docs, fix #69

jb-adams committed:
attempt to trigger build on travis-ci.com

mcupak committed:
Merge pull request #76 from gewenyu99/release-1.0.0

Merge in Hugo Docs

mcupak committed:
Clarify dependency on PrestoSQL, fix issue #66

mcupak committed:
Move travis-ci.org references to travis-ci.com, close #63

mcupak committed:
Merge pull request #80 from mcupak/issue/63

Move travis-ci.org references to travis-ci.com, close #63

mcupak committed:
Merge pull request #79 from mcupak/issue/66

Clarify dependency on PrestoSQL, fix issue #66

Commit:
fix top nav icons

Commit:
fixes fav icon

fixes fav icon

Commit:
Turn on draft for in progress pages

Commit:
un-ignre docs for deployment

Commit:
fix home page link in config

Commit:
update clyde to reflect span margin

Commit:
bump clyde version to fix header class name errors

Commit:
delete current unused file, add built docs back to ignore, and force add for deployment

Commit:
Local builds ignored, prod build command tracked for deployment.

adds warning message for running prod build

mcupak committed:
Merge pull request #88 from gewenyu99/release-1.0.0

Fixes stylistic issues in Hugo docs

mcupak committed:
Merge branch 'develop' into release-1.0.0

Commit:
fix #92 switches to tagged branch and only specify remote update on submodule.

Commit:
fix #85 bumps clyde version to allow titles to be displayed beside icon

Commit:
fix #91 moves installing client libraries to a more discrete location

Commit:
clean up loose ends with spelling and link errors

Commit:
issue #90 compile use case sections in readme

Commit:
issue #90, split use case examples into discovery and query

terminology change from schema discovery to table discovery

jfuerth committed:
Clarify: example page name is not prescriptive. fix #23

mcupak committed:
Merge pull request #100 from ga4gh-discovery/issue-23-clarify-pagination-example

Clarify: example page name is not prescriptive. fix #23

jfuerth committed:
Clarify that tables are arrays of JSON objects, not flat rectangles. Fix #97.

jfuerth committed:
Fix copy-paste error that marked 'id' as a person's age

Commit:
Use correct GA4GH Search name + Unify title case

jfuerth committed:
Removed claims that the API is REST; polished some wording (fix #99)

Commit:
update covid cloud examples to reflect table name changes

mcupak committed:
Merge pull request #110 from gewenyu99/develop

Update COVID Cloud examples to reflect table name changes

mcupak committed:
Merge pull request #106 from ga4gh-discovery/issue-99-rest

Removed claims that the API is REST; polished some wording (fix #99)

mcupak committed:
Merge #92

mcupak committed:
Fix build, split to jobs and set Node version, fix #115

mcupak committed:
Update .gitignore to keep current docs deployment

mcupak committed:
Rename Search to Data Connect, issue #112

Commit:
bump clyde version to remove search duplicates

Commit:
changed make clean to remove artifacts

Commit:
Updates readme with build instructions for hugo using the make file

mcupak committed:
Rename openapi file, issue #112

mcupak committed:
Fix inconsisten newline characters in USECASES.md

mcupak committed:
Fix typos in SEARCHSPEC.md

jfuerth committed:
Fix incorrect links

jfuerth committed:
Provide more detail in security section

mcupak committed:
Refactor README, improve summary and clarify scope, issue #111

mcupak committed:
Merge remote-tracking branch 'upstream/rebranding' into rebranding

mcupak committed:
Fix architecture image, issue #112

mcupak committed:
Fix typo in grammar name, issue #112

mcupak committed:
Minor change in phrasing in the use cases doc, issue #112

mcupak committed:
Incorporate Angela's comments, issue #111

mcupak committed:
Merge pull request #118 from ga4gh-discovery/rebranding

Rebranding

mcupak committed:
Rename repo, issue #112

mcupak committed:
Merge pull request #121 from ga4gh-discovery/repo-rename

Rename repo

mcupak committed:
Fix validation badge after renaming the repo, issue #112

mcupak committed:
Merge branch 'badge-rename' into develop

jfuerth committed:
Improve SchemaBlocks example. fix #96, ref #114

* corrects an error in the original schema definition
* references actual BloodGroup concept now submitted to SchemaBlocks playground
* adds example input data to make the example easier to follow

mcupak committed:
Merge pull request #122 from ga4gh-discovery/schemablocks-example-fix

Improve SchemaBlocks example. fix #96, ref #114

mcupak committed:
Rename a few remaining files and terms, fix #112

mcupak committed:
Rename Presto SQL to Trino, fix #78

mcupak committed:
Merge pull request #123 from mcupak/file-rename

Final housekeeping

jfuerth commented on USECASES.md:
```diff
@@ -1,117 +1,117 @@
+standard set of fields is "nice to have" or "must have", while the ability to
+have arbitrary non-standard fields is a "must have".
+
+Data type mentioned in the responses include:
```
Minor: "Data type" should be plural

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+      parameters:
+        - name: table_name
+          in: path
+          description: Unique Table name, unique to this implementation
```
Minor: this wording is a bit awkward, and it's repeated several times in this doc.

Maybe better to say "A table name"?

We could mention the unique thing on the `Table.name` property. We could also provide examples there such as `my_schema.my_table` and `my_catalog.my_schema.my_table` to make it clear that these are unique, qualified names.

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+  /table/{table_name}/data:
+    get:
+      summary: Fetch data from a Table
+      description: Optional operation that returns data from a Table
```
Is this one optional?

mcupak replied:
Huh no it shouldn't be.

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+              schema:
+                $ref: "#/components/schemas/TableData"
+        '400':
+          description: The query is invalid
```
Should we change this to "Error in request headers or body, for example if the query is invalid. Details are provided in the ErrorResponse body."?

mcupak replied:
Makes sense!

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+          description: |
+            Show information about this Data Connect service.
+
+            Use `"type": {"group": "org.ga4gh", "artifact": "data-connect", "version": "0.1.0"}` when implementing this specification directly.
```
Version should be "1.0.0"

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+        Describes a Table hosted by this Data Connect node.
+    TableData:
+      required:
+        - data_model
```
We should remove this. The `data_model` is only required here if the `data` array is present. This was a late change to allow servers to implement "status polling via pagination". Should also add description text about this.

jfuerth commented on spec/api.yaml:
```diff
@@ -0,0 +1,269 @@
+          example: 'Internal server error'
+      required:
+        - title
+    Pagination:
```
Add link to pagination rules in SPEC.md

jfuerth reviewed (commented):
Just a few loose ends that are probably worth cleaning up. I will send a new PR today to address my own comments here.

jfuerth committed:
A few fixes for release 1.0.0. ref #71

kemp-verily commented on hugo/content/docs/reference/pagination-long-queries.md:
```diff
@@ -0,0 +1,129 @@
+content-type: application/json
+retry-after: 1000
+
+{"data":[],"pagination":{"next_page_url":"/search/v1/statement/abc123/queued/1"}}
```
optional, but i think changing the URL here distracts a bit - there is no particular reason that the URL needs to change (this is a server implementation detail)

if you do want to show it, i would at least drop the queued/executing part and just keep the page numbers...

wdyt?

kemp-verily commented on hugo/content/docs/reference/sql-functions.md:
```diff
@@ -0,0 +1,190 @@
+---
+Data Connect's SQL dialect has been selected for compatibility with current major open source database platforms including Trino, PostgreSQL, and MySQL, and BigQuery. There are occasional name or signature differences, but a Data Connect implementation atop any of the major database platforms should be able to pass through queries that use the functions listed below with only minor tweaks.
+
+The functions below are a subset of those available in Trino 341. In a conformant Data Connect implementation, these functions must behave according to the Trino documentation. To assist with implementations directly on other database platforms, the [Trino Functions Support Matrix](https://docs.google.com/document/d/1y51qNuoe2ELX9kCOyQbFB4jihiKt2N8Qcd6-zzadIvk) captures the differences between pl ...
```
Is this net new or just moved?

I apologize if I missed the discussion but I am hesitant to make the claim that a conformant implementation needs to support everything here...

mcupak replied:
Not net new, should be just lifted from SPEC.md. I'll check.

mcupak replied:
I took a few minutes now to go over the list of functions, and I'm comfortable with the "optional" approach now.

kemp-verily commented on hugo/content/docs/security/data-source.md:
```diff
@@ -0,0 +1,77 @@
+
+This page discusses different approaches to securing a Data Connect implementation depending on which implementation path you choose and how complex your access needs are.
+
+Data Connect can be implemented in many ways:
```
`many ways.  For example:`

kemp-verily commented on hugo/content/docs/use-exisitng-data/tables-in-a-bucket/doc.md:
```diff
@@ -0,0 +1,53 @@
+type: docs
+layout: single-col
+---
+#### What is a Phenopacket?
```
Feels odd to open with this?  Can we move it down?

kemp-verily reviewed (commented):
some nits; still need to review a couple things; sorry for the delay

mcupak committed:
Merge pull request #125 from ga4gh-discovery/fixes-for-1.0

A few fixes for release 1.0.0.

mcupak commented:
@kemp-verily I agree with your comments. Are you going to send a PR or should I?

mcupak commented on SPEC.md:
```diff
@@ -0,0 +1,1592 @@
+
+With Trino (formerly PrestoSQL) being a popular database choice in Data Connect implementations, we've chosen its grammar as the basis for the grammar supported by Data Connect. Functions below are a subset of those available in Trino 341, and must behave according to the Trino documentation in a conforming Data Connect implementation. To assist with implementations directly on other database pla ...
+
+*   **Logical Operators**
```
This entire section seems to have broken formatting, we should fix this. Not sure what the significance of `*` at the end of some lines is, some identifiers are not escaped or backticked correctly, the final section is off and JSON functions has broken formatting.

mcupak replied:
Looks automated. @jfuerth Do you remember where we transferred this from?

Commit:
Modify the polling example URLs to be more opaque

The previous examples (in my opinion) pointed too much towards a particular
implementation.  This is an attempt to draw attention away from the actual URLs
and more towards the polling exchange.

Commit:
Remove obsolete pheno packet example

This depends on a private repository and is superceded by other documentation.

Commit:
Language cleanup

mcupak committed:
Merge pull request #129 from kemp-verily/fixes

Fixes for 1.0

kemp-verily committed:
Change stance on SQL dialect support (#130)

* Change stance on SQL dialect support

This greatly softens the language around SQL support.  A short list of
functions are specified as SHOULD implement, rather than a long list of MUST
implement.

The expectation is that we will revise this list over time as usage dictates.

This addresses issue #128.

mcupak commented:
Merging now that all the related issues have been resolved.
